archive_repo.py

```diff
@@ -0,0 +1,667 @@
+)
+# ...reviewFields definition includes ...commentFields and ...author
+# ...author definition includes ...userFields
+
+gql_PullRequest_Query = (
+    """
+      nodes { ...prFields }
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+      totalCount
+    }
+  }
+  rateLimit {
+    remaining
+    resetAt
+  }
+}
+"""
+    + gql_PullRequest_Fields
+)
+
+gql_AllPRs_Initial = (
+    """
+query($owner: String!, $repo: String!){
+  repository(owner: $owner, name: $repo) {
+    pullRequests(first: 10, orderBy: {field: UPDATED_AT, direction:DESC}) {
+"""
+    + gql_PullRequest_Query
+)
+
+gql_AllPRs_Subsequent = (
+    """
+query($owner: String!, $repo: String!, $cursor: String!){
+  repository(owner: $owner, name: $repo) {
+    pullRequests(first: 25, after: $cursor, orderBy: {field: UPDATED_AT, direction:DESC}) {
+"""
+    + gql_PullRequest_Query
+)
+
+gql_PR_Comments_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on PullRequest {
+            comments(first:100, after:$cursor) {
+                nodes { ...commentFields }
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
+}
+"""
+    + gql_Comment_Fields
+)
+
+gql_PR_Review_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on PullRequest {
+            reviews(first:100, after:$cursor) {
+                nodes { ...reviewFields }
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
+}
+"""
+    + gql_Review_Fields
+)
+
+gql_PR_ReviewComments_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on PullRequestReview {
+            comments(first: 50, after:$cursor) {
+                nodes {
+                    originalPosition
+                    ...commentFields
+                }
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
+}
+"""
+    + gql_Comment_Fields
+)
+
+##########################
+## Function definitions ##
+##########################
+
+
+last_request_limit = 5000
+
+
+def submit_query(query, variables, display):
+    global last_request_limit
+
+    url = "https://api.github.com/graphql"
+
+    bodyjson = {"query": re.sub("\s+", " ", query).strip()}
+    if variables:
+        bodyjson["variables"] = variables
+    body = json.dumps(bodyjson)
+
+    if not args.quiet:
+        output = f"Submitting query for {display} with "
+        output += str(variables) if variables else "no parameters"
+        print(output)
```

Comment:
Maybe this is worth defining:

```python
def eprint(*str, **kwargs):
    print(*str, file=sys.stderr, **kwargs)

if args.quiet:
    def log(**kwargs):
        pass
else:
    def log(*str, **kwargs):
        eprint(*str, **kwargs)
```

Writing progress messages to stderr and writing the archive to stdout seems like the nicest approach here, absent a `-o` argument.